pandas/io/tests/test_excel.py

```diff
@@ -423,7 +424,26 @@ def test_reader_converters(self):
         for path in (xls_path, xlsx_path):
             actual = read_excel(path, 'Sheet1', converters=converters)
             tm.assert_frame_equal(actual, expected)
-
+    
+    def test_reading_all_sheets(self):
+        # Test reading all sheetnames by setting sheetname to None,
+        # Ensure a dict is returned.
+        _skip_if_no_xlrd()
+        
+        dfs = read_excel(self.multisheet,sheetname=None)
+        expected_keys = ['Alpha','Beta','Charlie']
+        tm.assert_contains_all(expected_keys,dfs.keys())
+
+    def test_reading_multiple_specific_sheets(self):
+        # Test reading specific sheetnames by specifying a mixed list 
+        # of integers and strings.
+        # Ensure a dict is returned
+        _skip_if_no_xlrd()
+       
+        expected_keys = [2,'Charlie']
+        dfs = read_excel(self.multisheet,sheetname=expected_keys)
+        tm.assert_contains_all(expected_keys,dfs.keys())
+    
     def test_reader_seconds(self):
```

Comment:
I would rather have this tests, than relying on external sheets. Which maybe subtly different and non-reproducible. A complete end-to-end is simpler / more useful.